k4GaudiPandora/include/CalorimeterHitType.h

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2020-2024 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef CalorimeterHitType_h
+#define CalorimeterHitType_h 1
+
+#include <iostream>
+
+/** Helper class for decoding/encoding lcio::CalorimeterHit types for the ILD
+ *  detector. The encoding is: caloType + 10 * caloID + 1000 * layout + 10000 * layerNum <br>
+ *  (see enums: CaloType, CaloID and Layout for possible values).<br>
+ *  Example usage: <br>
+ *  <pre>
+ *     lcio::CalorimeterHit* cHit = .... ;
+ *
+ *     // set the type (e.g. in digitization )
+ *     cHit->setType( CHT( CHT::em ,  CHT::ecal , CHT::plug , 12 ) ) ;
+ *
+ *     ...
+ *
+ *     CHT cht = cHit->getType() ;
+ *
+ *     //   sum energies for electromagentic, hadronic and tailcatcher:
+ *     if( cht.is( CHT::em ) )
+ *          e_em +=  cHit->getEnergy() ;
+ *     else
+ *       if ( cht.is(CHT::had ) )
+ *          e_had += cHit->getEnergy() ;
+ *       else
+ *          e_muon += cHit->getEnergy() ;
+ *
+ *     // use only EcalPlug hits:
+ *     if( cht.is( CHT::ecal) && cht.is( CHT::plug) )
+ *
+ *     // get the layer number (e.g. for calibration or clustering)
+ *     unsigned l = cht.layer() ;
+ *     // or directly :
+ *     unsigned l = CHT(  cHit->getType() ).layer()  ;
+ *
+ *     // detailed print:
+ *     std::cout <<  CHT(  cHit->getType() ) << std::endl ;
+ *
+ *  </pre>
+ *
+ *  F.Gaede, DESY, 12/2008
+ */
+
+class CHT {
+public:
+  /** calorimeter types */
+  typedef enum { em = 0, had = 1, muon = 2 } CaloType;
+
+  /** calo ids - specific to ILD */
+  typedef enum { unknown = 0, ecal = 1, hcal = 2, yoke = 3, lcal = 4, lhcal = 5, bcal = 6 } CaloID;
+
+  /** calo layout / subdetector  */
+  typedef enum { any = 0, barrel = 1, endcap = 2, plug = 3, ring = 4 } Layout;
```

Comment:
```suggestion
   enum class CaloType { em = 0, had = 1, muon = 2 };

  /** calo ids - specific to ILD */
  enum class CaloID { unknown = 0, ecal = 1, hcal = 2, yoke = 3, lcal = 4, lhcal = 5, bcal = 6 };

  /** calo layout / subdetector  */
  enum class Layout { any = 0, barrel = 1, endcap = 2, plug = 3, ring = 4 };
```
Take this opportunity to modernize the whole thing a bit. (This will most likely require work in a few places).

Additionally, these enums are fairly specific for ILD, I don't know if they will work for all concepts. But that is probably something to worry about at a later point.

Reply:
Even better if we could get rid of CHT; in `DDMarlinPandora` it's used only in a few places and it's specific to ILD. I think we don't use many of whatever is in the files anyway.